<a href="https://colab.research.google.com/github/tevfikcagridural/student-success/blob/master/Data_Wrangling_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Imports

In [7]:
from google.colab import drive
import pandas as pd

## Setting up Google Drive connection

In [8]:
drive.mount('/gdrive')

Mounted at /gdrive


## Importing the data

In [10]:
data_folder =  '/gdrive/My Drive/Data Science Career Track/Unit 7/capstone_two_project_folder/data/'

In [11]:
file_assessments  = data_folder + 'assessments.csv'
file_courses = data_folder + 'courses.csv'
file_student_assessment = data_folder + 'studentAssessment.csv'
file_student_info = data_folder + 'studentInfo.csv'
file_student_register = data_folder + 'studentRegistration.csv'
file_student_vle = data_folder + 'studentVle.csv'
file_vle = data_folder + 'vle.csv'

In [12]:
assessments = pd.read_csv(file_assessments)
courses = pd.read_csv(file_courses)
student_assessment = pd.read_csv(file_student_assessment)
student_info = pd.read_csv(file_student_info)
student_register = pd.read_csv(file_student_register)
student_vle = pd.read_csv(file_student_vle)
vle = pd.read_csv(file_vle)

## Eyeballing on the data

In [13]:
assessments.head()

code_module code_presentation  id_assessment assessment_type   date  weight
0         AAA             2013J           1752             TMA   19.0    10.0
1         AAA             2013J           1753             TMA   54.0    20.0
2         AAA             2013J           1754             TMA  117.0    20.0
3         AAA             2013J           1755             TMA  166.0    20.0
4         AAA             2013J           1756             TMA  215.0    30.0

In [14]:
courses.head()

code_module code_presentation  module_presentation_length
0         AAA             2013J                         268
1         AAA             2014J                         269
2         BBB             2013J                         268
3         BBB             2014J                         262
4         BBB             2013B                         240

In [15]:
student_assessment.head()

id_assessment  id_student  date_submitted  is_banked  score
0           1752       11391              18          0   78.0
1           1752       28400              22          0   70.0
2           1752       31604              17          0   72.0
3           1752       32885              26          0   69.0
4           1752       38053              19          0   79.0

In [16]:
student_info.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32593 entries, 0 to 32592
Data columns (total 12 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   code_module           32593 non-null  object
 1   code_presentation     32593 non-null  object
 2   id_student            32593 non-null  int64 
 3   gender                32593 non-null  object
 4   region                32593 non-null  object
 5   highest_education     32593 non-null  object
 6   imd_band              31482 non-null  object
 7   age_band              32593 non-null  object
 8   num_of_prev_attempts  32593 non-null  int64 
 9   studied_credits       32593 non-null  int64 
 10  disability            32593 non-null  object
 11  final_result          32593 non-null  object
dtypes: int64(3), object(9)
memory usage: 3.0+ MB


In [17]:
student_info.head()

code_module code_presentation  ...  disability final_result
0         AAA             2013J  ...           N         Pass
1         AAA             2013J  ...           N         Pass
2         AAA             2013J  ...           Y    Withdrawn
3         AAA             2013J  ...           N         Pass
4         AAA             2013J  ...           N         Pass

[5 rows x 12 columns]

In [18]:
student_register.head()

code_module code_presentation  ...  date_registration  date_unregistration
0         AAA             2013J  ...             -159.0                  NaN
1         AAA             2013J  ...              -53.0                  NaN
2         AAA             2013J  ...              -92.0                 12.0
3         AAA             2013J  ...              -52.0                  NaN
4         AAA             2013J  ...             -176.0                  NaN

[5 rows x 5 columns]

In [19]:
student_vle.head()

code_module code_presentation  id_student  id_site  date  sum_click
0         AAA             2013J       28400   546652   -10          4
1         AAA             2013J       28400   546652   -10          1
2         AAA             2013J       28400   546652   -10          1
3         AAA             2013J       28400   546614   -10         11
4         AAA             2013J       28400   546714   -10          1

In [31]:
student_vle = student_vle.set_index(keys=['code_module', 'code_presentation'])

In [39]:
student_vle.groupby(by='id_student').describe()

id_site                                         ... sum_click                
             count           mean           std       min  ...       25%  50%  75%   max
id_student                                                 ...                          
6516         662.0  877282.676737   5179.638978  877011.0  ...       1.0  2.0  5.0  49.0
8462         304.0  675792.838816  15970.728853  673519.0  ...       1.0  1.0  2.0  16.0
11391        196.0  546697.668367     99.183214  546614.0  ...       1.0  2.0  5.0  76.0
23629         59.0  542863.067797     68.365111  542797.0  ...       1.0  2.0  3.0  13.0
23698        305.0  911160.937705  10922.412711  909013.0  ...       1.0  1.0  2.0  78.0
...            ...            ...           ...       ...  ...       ...  ...  ...   ...
2698251      529.0  773040.007561    237.075721  772700.0  ...       1.0  2.0  3.0  22.0
2698257      267.0  546690.715356     94.421316  546614.0  ...       1.0  2.0  3.0  16.0
2698535      899.0  588821.272525  73026.539870  551010.0  ...       1.0  2.0  5.0  78.0
2698577      237.0  920655.780591  19387.791769  913466.0  ...       1.0  2.0  3.0  40.0
2698588      226.0  917075.752212  12897.866752  913466.0  ...       1.0  1.5  3.0  19.0

[26074 rows x 24 columns]

In [20]:
vle.head()

id_site code_module code_presentation activity_type  week_from  week_to
0   546943         AAA             2013J      resource        NaN      NaN
1   546712         AAA             2013J     oucontent        NaN      NaN
2   546998         AAA             2013J      resource        NaN      NaN
3   546888         AAA             2013J           url        NaN      NaN
4   547035         AAA             2013J      resource        NaN      NaN

In [21]:
vle.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6364 entries, 0 to 6363
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id_site            6364 non-null   int64  
 1   code_module        6364 non-null   object 
 2   code_presentation  6364 non-null   object 
 3   activity_type      6364 non-null   object 
 4   week_from          1121 non-null   float64
 5   week_to            1121 non-null   float64
dtypes: float64(2), int64(1), object(3)
memory usage: 298.4+ KB


## Data Merge
I will merge all the tables as each row is for each student. However, `num_of_prev_attemts` colums rises a question here.  
  
Should each row considered as another person or their former interaction informations whould be added on their final attempt.
Eventhough the second option sounds more logical data does not provide any information about relationships between rows. So, basically even if their `num_of_revattmepts > 1` we don't know which one is the former attempt. More, it can be that the students 2nd attempt -if they took it before 2013J- and the former attempts may not be in the data. An example `ìd_student = 23629`

In [22]:
pd.concat?